Atomic/rpmwriter.py

```diff
@@ -0,0 +1,369 @@
+import os
+import sys
+import struct
+import subprocess
+import select
+import gzip
+import hashlib
+import tempfile
+import shutil
+
+# More information on the RPM file format can be found here:
+# http://ftp.rpm.org/max-rpm/s1-rpm-file-format-rpm-file-format.html
+# The same documentation was used to initially write this module.
+#
+# Only tags that are actually used in the code were added to the RpmWriter class.
+# The complete list is available here:
+# https://github.com/rpm-software-management/rpm/blob/master/lib/rpmtag.h
+
+class RpmWriter(object):
+
+    MAGIC = [0xed, 0xab, 0xee, 0xdb]
+    MAJOR = [0x3]
+    MINOR = [0x0]
+    BINARY_TYPE = [0x0, 0x0]
+    NOARCH = [0xFF, 0xFF]
+    OS = [0x0, 0x1]
+    SIGNATURE_TYPE = [0x0, 0x5]
+    RESERVED = [0x0] * 16
+
+    HEADER_MAGIC = [0x8e, 0xad, 0xe8]
+    HEADER_VERSION = [0x01]
+    HEADER_RESERVED = [0x0] * 4
+
+    SIGNATURE_MAGIC = [0x8e, 0xad, 0xe8]
+    SIGNATURE_VERSION = [0x01]
+    SIGNATURE_RESERVED = [0x0] * 4
+
+    RPMSIGTAG_PAYLOADSIZE = 1007
+
+    RPMTAG_NAME = 1000
+    RPMTAG_VERSION = 1001
+    RPMTAG_RELEASE = 1002
+    RPMTAG_SUMMARY = 1004
+    RPMTAG_DESCRIPTION = 1005
+    RPMTAG_LICENSE_ = 1014
+    RPMTAG_GROUP = 1016
+    RPMTAG_CHANGELOG = 1017
+    RPMTAG_URL = 1020
+    RPMTAG_OS = 1021
+    RPMTAG_ARCH = 1022
+    RPMTAG_FILESIZES = 1028
+    RPMTAG_DIRINDEXES = 1116
+    RPMTAG_BASENAMES = 1117
+    RPMTAG_DIRNAMES = 1118
+    RPMTAG_FILEMODES = 1030
+    RPMTAG_FILERDEVS = 1033
+    RPMTAG_FILEMTIMES = 1034
+    RPMTAG_FILEDIGESTS = 1035
+    RPMTAG_FILELINKTOS = 1036
+    RPMTAG_FILEFLAGS = 1037
+    RPMTAG_FILEUSERNAME = 1039
+    RPMTAG_FILEGROUPNAME = 1040
+
+    RPMTAG_PROVIDENAME		= 1047
+    RPMTAG_REQUIRENAME		= 1049
+    RPMTAG_REQUIREVERSION	= 1050
+    RPMTAG_CONFLICTFLAGS	= 1053
+    RPMTAG_CONFLICTNAME		= 1054
```

Comment:
micro nit.  You've an extra tab it looks like between the variable and the equals for a few of these definitions.    I personally like this style and would ask that you do the same for the group of variables above too (lines 40->62)